テーブル設計

yokoe9073 opened the pull request:
<WHAT>
READMEのコード記入
ER図を作成→https://drive.google.com/file/d/1NMzxs-B96dYwFD-uSUjcv8Fna0iTxxOZ/view?usp=sharing

<WHY>
README はアプリケーションの仕様を記載するため
ER図はテーブル間の関係を把握しやすくするために作成

yokoe9073 committed:
gem rails_12factorの追加

yokoe9073 committed:
Basic認証を導入

yokoe9073 committed:
README設計

isyt729 commented:
レビュー依頼ありがとうございます！
こちらのプルリクエストですが、プルリクエストにWhatとWhyが書かれていません。
What（どのような実装をしているのか）と Why（なぜこの実装が必要なのか）を書くことで、
このブランチは何を実装していて、なぜその実装を行ったのかが他の人に伝えることができます。
What をしっかり書くことで、他者にコードを見てもらう際に、スムーズな理解を促すことができ、Whyを書くことでなぜこの実装を行ったのか、本来の目的に沿ったレビュアーとのコミュニケーションが可能となります。
お手数かけますが、記入後再度提出いただけますよう、よろしくお願いします。
https://master.tech-camp.in/curriculums/4675

isyt729 commented on README.md:
```diff
@@ -1,24 +1,59 @@
-Things you may want to cover:
+| Column   | Type   | Options     |
+| -------- | ------ | ----------- |
+| name     | string | null: false |
```
```
| name     | string | null: false |
```
ニックネームと本名の姓名、それぞれの振り仮名を分けて保存できるようにしましょう！

isyt729 commented on README.md:
```diff
@@ -1,24 +1,59 @@
+| name     | string | null: false |
+| email    | string | null: false |
+| password | string | null: false |
 
```
usersテーブルは必要なカリキュラムが不足しているので、確認しましょう！

isyt729 commented on README.md:
```diff
@@ -1,24 +1,59 @@
+| ---------------------| ----------| ------------------------------|
+| item_name            | string    | null: false                   |
+| image                | string    | null: false                   |
+| category             | string    | null: false                   |
+| price                | integer   | null: false                   | 
+| status               | string    | null: false                   |
+| delivery_fee         | integer   | null: false                   |
+| exhibitor_prefectures| string    | null: false                   |
+| shipping_date        | string    | null: false                   |
```
```
| category             | string    | null: false                   |
| status               | string    | null: false                   |
| exhibitor_prefectures| string    | null: false                   |
| shipping_date        | string    | null: false                   |
```
こちらはactive_hashを用いるので、integerにしましょう！

isyt729 commented on README.md:
```diff
@@ -1,24 +1,59 @@
+- has_one :buyer
 
-* Configuration
+## items テーブル
```
itemsテーブルに必要なカラムが不足しているので、確認しましょう！

isyt729 commented on README.md:
```diff
@@ -1,24 +1,59 @@
-* Database initialization
+### Association
 
-* How to run the test suite
```
buyerテーブルとのアソシエーションが必要なので、追加しましょう！
また、buyerテーブル側にも必要なアソシエーション、カラムを追加しましょう！

isyt729 commented on README.md:
```diff
@@ -1,24 +1,59 @@
+| Column           | Type       | Options                        |
+| -----------------| ---------- | -------------------------------|
+| user             | references | null: false, foreign_key: true |
+| shopping_address | string     | null: false                    |
```
addressは複数のカラムに分けてデータを保存する必要があります。
テーブルを作成し、設計を見直しましょう！

isyt729 reviewed (commented):
レビュー依頼ありがとうございます！
コメントしましたので、ご確認お願いします！

yokoe9073 committed:
README修正6箇所

①ニックネーム、姓名、フリガナ追加②usersテーブルのカラムを追加③active_hashを用いる部分をintegerに変更④itemsテーブルにカラム追加⑤buyerテーブルにアソシエーション、カラム追加⑥shippingテーブル追加

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,75 @@
+## items テーブル
+| Column               | Type      | Options                       |
+| ---------------------| ----------| ------------------------------|
+| item_name            | string    | null: false                   |
```
こちらテーブル名から何の名前か判断できるので、nameで大丈夫です！

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,75 @@
+| Column           | Type       | Options                        |
+| -----------------| ---------- | -------------------------------|
+| user             | references | null: false, foreign_key: true |
+| card_info        | string     | null: false                    |
```
セキュリティの観点から、クレジットカードの情報はデータベースに直接保存しないようにしましょう。

また、今回のフリマアプリではカードを登録する際は購入の度に入力となり、データベースにわざわざ記録しなくても良い情報となります。
ですので、これらのカラムを用意しなくても大丈夫です。

yokoe9073 replied:
ご指摘ありがとうございました
指摘いただいた2箇所を修正しました

yokoe9073 committed:
README修正2箇所

①itemsテーブルにてnameへ変更②カード情報のカラムを削除

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,74 @@
 
-This README would normally document whatever steps are necessary to get the
-application up and running.
+## users テーブル
```
誕生日を入れるカラムを作成してあげましょう！

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,74 @@
+### Association
+- has_many :items
+- has_many :comments
+- has_one :buyer
```
こちらhas_manyですね！　`has_one :buyer`

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,74 @@
+## buyer テーブル
+| Column           | Type       | Options                        |
+| -----------------| ---------- | -------------------------------|
+| user             | references | null: false, foreign_key: true |
```
itemの外部キー制約もつけてあげましょう！

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,74 @@
+| city       | string    | null: false                    |
+| address    | string    | null: false                    | 
+| building   | string    | null: false                    |
+| tel        | integer   | null: false                    |
```
こちら、電話番号は数字の文字列として実装しておきたいためstring型で実装しましょう！

yokoe9073 replied:
ご指摘ありがとうございます
４箇所の修正と、buyerテーブルにshippingカラムを追加しました。
ご確認をお願いします

yokoe9073 committed:
README修正4箇所+@

①usersテーブルに誕生日カラムを追加②usersテーブルのアソシエーションにてhas_many :buyerに変更③buyerテーブルにitemのカラム追加④telのカラムの型をstringへ変更 @buyerテーブルにてshippingカラムを追加

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,77 @@
+| nickname     | string | null: false |
+| email        | string | null: false |
+| password     | string | null: false |
+| birthday     | integer| null: false |
```
誕生日カラムは「date型」を使いましょう。

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,77 @@
+| -----------------| ---------- | -------------------------------|
+| user             | references | null: false, foreign_key: true |
+| item             | references | null: false, foreign_key: true |
+| shipping         | references | null: false, foreign_key: true |
```
こちらは不要ですね！
`| shipping         | references | null: false, foreign_key: true |`

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,77 @@
+| prefectures| integer   | null: false                    |
+| city       | string    | null: false                    |
+| address    | string    | null: false                    | 
+| building   | string    | null: false                    |
```
建物番号(建物名)の入力は任意になるため、not null制約はなくても大丈夫です!

s-aka-moto replied:
こちらnull:falseを消してあげましょう！
`| building   | string    | null: false                    |`

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,77 @@
+| item             | references | null: false, foreign_key: true |
+| shipping         | references | null: false, foreign_key: true |
+### Association
+- belongs_to :users
+- belongs_to :items
```
belongs_toは単数形ですね！

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,77 @@
+| user          | references | null: false, foreign_key: true |
+| item          | references | null: false, foreign_key: true |
+### Association
+- belongs_to :users
+- belongs_to :items
```
belongs_toは単数形ですね！

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,77 @@
+| shipping_date        | integer   | null: false                   |
+| user                 | references| null: false, foreign_key: true|
+### Association
+- belongs_to :users
```
belongs_toは単数形ですね！

yokoe9073 committed:
README修正６箇所

①誕生日カラムの型をdateに変更　②shippingカラムの削除③建物のnot null制約削除　④⑤⑥belong_toの後を単数形に変更

yokoe9073 committed:
README修正１箇所

buildingカラムのoptionからnull: falseを削除しました
